src/soroban_rpc.ts

```diff
@@ -162,16 +162,90 @@ export namespace SorobanRpc {
     retval: xdr.ScVal;
   }
 
-  export interface SimulateTransactionResponse {
+  /**
+   * Simplifies {@link RawSimulateTransactionResponse} into separate interfaces
+   * based on status:
+   *   - on success, this includes all fields, though `result` is only present
+   *     if an invocation was simulated (since otherwise there's nothing to
+   *     "resultify")
+   *   - if there was an expiration error, this includes error and restoration
+   *     fields
+   *   - for all other errors, this only includes error fields
+   *
+   * @see https://soroban.stellar.org/api/methods/simulateTransaction#returns
+   */
+  export type SimulateTransactionResponse =
+    | SimulateTransactionSuccessResponse
+    | SimulateTransactionRestoreResponse
+    | SimulateTransactionErrorResponse;
+
+  interface BaseSimulateTransactionResponse {
+    /** always present: the JSON-RPC request ID */
     id: string;
-    error?: string;
-    transactionData: SorobanDataBuilder;
+
+    /** always present: the LCL known to the server when responding */
+    latestLedger: number;
+
+    /**
+     * The field is always present, but may be empty in cases where:
+     *   - you didn't simulate an invocation,
+     *   - there were no events, or
+     *   - you are communicating with an RPC server w/o diagnostic events on
```

Comment:
I think can remove ln 193 in terms of events with simulations, as should always get diag events back from simulation, because diag level is hardcoded in the preflight rpc endpoint when it configs host lib - https://github.com/stellar/soroban-tools/blob/main/cmd/soroban-rpc/lib/preflight/src/preflight.rs#L51

it looks like it doesn't go through ingestion with captive core in rpc which is where diag events could be (dis)abled through captive .cfg, I hope I'm stating this accurately.